account_payment_partner/models/account_invoice.py

```diff
@@ -27,18 +27,19 @@ def _onchange_partner_id(self):
                     force_company=self.company_id.id
                 ).partner_id.supplier_payment_mode_id
                 self.payment_mode_id = pay_mode
+                self.partner_bank_id = False
                 if (
                     pay_mode and
                     pay_mode.payment_type == 'outbound' and
                     pay_mode.payment_method_id.bank_account_required and
                     self.commercial_partner_id.bank_ids
                 ):
-                    self.partner_bank_id = \
+                    partner_banks = \
                         self.commercial_partner_id.bank_ids.filtered(
                             lambda b: b.company_id == self.company_id or not
                             b.company_id)
-                else:
-                    self.partner_bank_id = False
+                    if partner_banks:
```

Comment:
No need to make `if`. If you put at the end `[:1]`, it servers for 0 or more elements to filter the first one.